### PR TITLE
Implement emitter handle MsgEditDataSource/OracleScript.

### LINKED ISSUE #2070: Edit data source/oracle script on DB
## Goal
The database must be updated changed fields follow what happens in chain.

### PR DESCRIPTION
**Pull request by Benzbeeb**
Fixed: #2070 

## Implementation details
1. Implement function set data source/oracle script for sending msg to Kafka.
2. Implement set_data_source/oracle_script handler for flusher.
3. Implement handle msgEditDataSource/OracleScript.
---

Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [x] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [x] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: chain/emitter/oracle.go [hunk 1]
```diff
@@ -5,6 +5,30 @@ import (
 	"github.com/bandprotocol/bandchain/chain/x/oracle/types"
 )
 
+func (app *App) SetDataSource(id types.DataSourceID, ds types.DataSource, txHash []byte) {
```
Comment (sorawit) [inline chain/emitter/oracle.go]: lowercase the funciton. Similar to `emitSetValidator`. Please be consistent.

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/emitter/oracle.go [hunk 2]
```diff
@@ -88,16 +106,27 @@ func (app *App) handleMsgCreateOracleScript(
...
+}
+
+// handleMsgEditDataSource implements emitter handler for MsgEditDataSource.
+func (app *App) handleMsgEditDataSource(
+	txHash []byte, msg oracle.MsgEditDataSource, evMap EvMap, extra JsDict,
+) {
+	id := types.DataSourceID(atoi(evMap[types.EventTypeEditDataSource+"."+types.AttributeKeyID][0]))
+	ds := app.BandApp.OracleKeeper.MustGetDataSource(app.DeliverContext, id)
+	app.emitSetDataSource(id, ds, txHash)
+	extra["id"] = id
```
Comment (sorawit) [inline chain/emitter/oracle.go]: remove extra. It's not needed.

File: chain/emitter/oracle.go [hunk 3]
```diff
@@ -88,16 +106,27 @@ func (app *App) handleMsgCreateOracleScript(
...
-	})
+	app.emitSetOracleScript(id, os, txHash)
+	extra["id"] = id
+}
+
+// handleMsgEditDataSource implements emitter handler for MsgEditDataSource.
+func (app *App) handleMsgEditDataSource(
+	txHash []byte, msg oracle.MsgEditDataSource, evMap EvMap, extra JsDict,
+) {
+	id := types.DataSourceID(atoi(evMap[types.EventTypeEditDataSource+"."+types.AttributeKeyID][0]))
```
Comment (sorawit) [inline chain/emitter/oracle.go]: Can just use msg.id directly.

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by sorawit — APPROVED** [review]

File: chain/emitter/handler.go [hunk 4]
```diff
@@ -34,6 +34,10 @@ func (app *App) handleMsg(txHash []byte, msg sdk.Msg, log sdk.ABCIMessageLog, ex
 		app.handleMsgCreateDataSource(txHash, msg, evMap, extra)
 	case oracle.MsgCreateOracleScript:
 		app.handleMsgCreateOracleScript(txHash, msg, evMap, extra)
+  case oracle.MsgEditDataSource:
```
Comment (taobun) [inline chain/emitter/handler.go]: Boo!

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by taobun — APPROVED** [review]

**Pull request merged**